build_cmake/helpers.cmake

```diff
@@ -579,3 +579,55 @@ function(assert_type_size type size)
         message(FATAL_ERROR "Type assertion failed: ${type} does not equal size ${size}")
     endif()
 endfunction()
+
+# parse_filelist_source(filelist output_var)
+# A helper function that parses the list of sources (usually found in "dist/filelist"). This returning a list of
+# sources that can then be used to generate the necessary build rules for the wiredtiger library. Additionally
+# uses the config values "WT_ARCH" and "WT_OS" when extracting platform specific sources from the file list.
+#   filelist - Destination of 'filelist' file.
+#   output_var - name of the output variable that will be set with the parsed sources. Output variable is set in
+#       the callers scope.
+function(parse_filelist_source filelist output_var)
+    set(arch_host "")
+    set(plat_host "")
+    # Determine architecture host for our filelist parse
+    if(WT_X86)
+        set(arch_host "X86_HOST")
+    elseif(WT_ARM64)
+        set(arch_host "ARM64_HOST")
+    elseif(WT_PPC64)
+        set(arch_host "POWERPC_HOST")
+    elseif(WT_ZSERIES)
+        set(arch_host "ZSERIES_HOST")
+    endif()
+    # Determine platform host for our filelist parse
+    if(WT_POSIX)
+        set(plat_host "POSIX_HOST")
+    elseif(WT_WIN)
+        set(plat_host "WINDOWS_HOST")
+    endif()
+
+    # Read file list and parse into list
+    file(READ "${filelist}" contents NEWLINE_CONSUME)
+    string(REGEX REPLACE "\n" ";" contents "${contents}")
+    set(output_files "")
+    foreach(file ${contents})
+        if(${file} MATCHES "^#.*$")
+            continue()
+        endif()
+        string(REGEX REPLACE "[ \t\r\]+" ";" file_contents ${file})
```

Comment:
What does the last `\` in the square bracket match? Is it actually a `\n`?

Reply:
Good find! The extra backslash is not necessary and has no effect. I've removed it.